pytest_fauxfactory/marks.py

```diff
@@ -27,6 +27,12 @@ def faux_string(items, str_type=None, *args, **kwargs):
     if str_type is None:
         str_type = fauxfactory.gen_choice(STRING_TYPES)
 
+    kwargs['length'] = kwargs.get('length', [None])
+    if not isinstance(kwargs['length'], list):
+        kwargs['length'] = [kwargs['length']]
+    length_cycle = cycle(kwargs['length'])
+
     while item < items:
+        kwargs['length'] = next(length_cycle)
```

Comment:
I see `kwargs['length']` getting set multiple times, so I think there is room for some refactoring here.

Reply:
The best i can think of right now would be to change lines 30-34 to this:
```python
if not isinstance(kwargs.get('length', None), list):
    kwargs['length'] = [kwargs.get('length', None)]
length_cycle = cycle(kwargs['length'])
```
Two checks need to be made. First, that the length key exists. Second, that the value is in iterable. If either of those are untrue, it needs to be put in a state where they are true. The above version is a bit wonky, but reduces the code by one line and only sets the variable once.
All tests still passing
```python
tests/test_faux_string.py::test_gen_alpha_string_with_variable_length[VgHjF] PASSED                                                                                                                                          [ 96%]
tests/test_faux_string.py::test_gen_alpha_string_with_variable_length[HguucpitYAZATHv] PASSED                                                                                                                                [ 97%]
tests/test_faux_string.py::test_gen_alpha_string_with_variable_length[BDJMR] PASSED                                                                                                                                          [ 98%]
tests/test_faux_string.py::test_gen_alpha_string_with_variable_length[dGfXuPaQqyPrrnR] PASSED  
```
If that works for you, I can push up the changes. This can also include the multi-string method code as well. Though I still don't know of a clever way to test it,

Reply:
Alright. So I've come up with a clever way to test the multi-string types! If the change I suggested above works for you, then I can push up the amended changes.

Reply:
Sorry for the late reply. I meant to review it again and I made the mistake of looking at work email first :/ I'll get back to you Monday but please go ahead with all your changes and rebase?